### ISSUE TITLE
[java] Running mvn site command excludes high priority violations

### ISSUE THREAD

**Issue by lucasmrMM** [pmd/pmd#4086]
**Affects PMD Version:** 6.46.0

**Description:**

When running ``mvn clean install site`` over a Java Maven project, having the ``<failOnViolation>false</failOnViolation>`` plugin configuration, some violations displayed in the console output before the testing phase are missing during Maven report site generation.

**Additional tech information:**
- JDK version: Openjdk version "11.0.12" 2021-07-20
- Maven version: Apache Maven 3.8.6 (84538c9988a25aec085021c365c560670ad80f63)
- Wndows 10 + WSL 2 (Windows Subsystem for Linux 2)
  - Linux Distribution: Ubuntu-20.04

**Code Sample demonstrating the issue:**

In order to trigger the scenario, you will need to add some piece of code to the main sources for analysis. You need to add a 'switch' statement without the ``default`` clause and without any ``break`` statement. Take the example below:

```
    switch (type) {
        case ENTERPRISE:
            validator = new Validator();
    }
```

**Steps to reproduce:**

Please provide detailed steps for how we can reproduce the bug.

1.  📐 Create a custom rule set XML file with the following:

```
<?xml version="1.0"?>

<ruleset name="Testing ruleset"
         xmlns="http://pmd.sourceforge.net/ruleset/2.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 https://pmd.sourceforge.io/ruleset_2_0_0.xsd">

    <rule ref="category/java/errorprone.xml/ImplicitSwitchFallThrough"/>
    <rule ref="category/java/bestpractices.xml/SwitchStmtsShouldHaveDefault">
        <priority>1</priority>
    </rule>

</ruleset>
```

2.  📝 Prepare your POM file with the following:

```
    <properties>
        <pmd-maven-plugin.version>3.17.0</pmd-maven-plugin.version>
        
        <!-- ⚠️⚠️Replace with the path to your ruleset XML file⚠️⚠️ -->
        <pmd.rules-path>${reporting.rules-directory}/pmd_rules.xml</pmd.rules-path>
    </properties>

    <build>
        <plugins>
            <!-- Maven Site Plugin -->
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-site-plugin</artifactId>
                <version>3.12.0</version>
                <configuration>
                    <relativizeDecorationLinks>false</relativizeDecorationLinks>
                </configuration>
            </plugin>

            <!-- Surefire -->
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-surefire-plugin</artifactId>
                <version>2.22.2</version>
            </plugin>

            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-pmd-plugin</artifactId>
                <version>${pmd-maven-plugin.version}</version>
                <configuration>
                    <printFailingErrors>true</printFailingErrors>
                    <failOnViolation>false</failOnViolation>
                    <rulesets>
                        <ruleset>${pmd.rules-path}</ruleset>
                    </rulesets>
                </configuration>
                <executions>
                    <execution>
                        <id>validate</id>
                        <phase>validate</phase>
                        <goals>
                            <goal>check</goal>
                            <goal>cpd-check</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
        </plugins>
    </build>

    <reporting>
        <plugins>
            <!-- Maven Project Info Reports Plugin -->
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-project-info-reports-plugin</artifactId>
                <version>2.9</version>
                <configuration>
                    <dependencyLocationsEnabled>false</dependencyLocationsEnabled>
                </configuration>
                <reportSets>
                    <reportSet>
                        <reports>
                            <report>index</report>
                            <report>summary</report>
                        </reports>
                    </reportSet>
                </reportSets>
            </plugin>
            
            <!-- Surefire Report -->
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-surefire-report-plugin</artifactId>
                <version>2.22.2</version>
            </plugin>
            
            <!-- JXR: Make it easier for the user to reference or find specific lines of code in the report -->
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-jxr-plugin</artifactId>
                <version>2.5</version>
                <reportSets>
                    <reportSet>
                        <reports>
                            <!-- Only generate references to "src" .java files, and excludes "test" .java files -->
                            <report>jxr</report>
                        </reports>
                    </reportSet>
                </reportSets>
            </plugin>
            
            <!-- PMD -->
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-pmd-plugin</artifactId>
                <version>${pmd-maven-plugin.version}</version>
                <configuration>
                    <rulesets>
                        <ruleset>${pmd.rules-path}</ruleset>
                    </rulesets>
                </configuration>
            </plugin>
        </plugins>
    </reporting>
```

3. ⌨️ Open a terminal in the root directory of the Maven project and run the following: ``mvn clean install site``

4. 🖥️ The console output will display the following:

```
[INFO] >>> maven-pmd-plugin:3.17.0:check (validate) > :pmd @ foo >>>
[INFO]                                                                    
[INFO] --- maven-pmd-plugin:3.17.0:pmd (pmd) @ foo ---              
[INFO] PMD version: 6.46.0
[INFO] 
[INFO] <<< maven-pmd-plugin:3.17.0:check (validate) < :pmd @ foo <<<
[INFO]                                                                    
[INFO]
[INFO] --- maven-pmd-plugin:3.17.0:check (validate) @ foo ---

⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️
      LOOK AT THE FOLLOWING 4 LINES!!
⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️

[INFO] PMD version: 6.46.0
[INFO] PMD Failure: com.foo.ExampleClass:17 Rule:ImplicitSwitchFallThrough Priority:3 A switch statement does not contain a break.
[INFO] PMD Failure: com.foo.ExampleClass:17 Rule:SwitchStmtsShouldHaveDefault Priority:1 Switch statements should be exhaustive, add a default c
ase (or missing enum branches).
[INFO] You have 2 PMD violations. For more details see: /[PROJECT-DIRECTORY]/target/pmd.xml
[INFO]
[INFO] >>> maven-pmd-plugin:3.17.0:cpd-check (validate) > :cpd @ foo >>>
[INFO]
[INFO] --- maven-pmd-plugin:3.17.0:cpd (cpd) @ foo ---
[INFO] PMD version: 6.46.0
[INFO] 
[INFO] <<< maven-pmd-plugin:3.17.0:cpd-check (validate) < :cpd @ foo <<<
[INFO]
[INFO]
[INFO] --- maven-pmd-plugin:3.17.0:cpd-check (validate) @ foo ---
[INFO] PMD version: 6.46.0
[INFO]
[INFO]

...

[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running com.foo.ExampleClassTest
[INFO] Tests run: 28, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.619 s 
[INFO] 
[INFO] Results:
[INFO]
[INFO] Tests run: 28, Failures: 0, Errors: 0, Skipped: 0
[INFO]

...

[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ foo ---
...
[INFO] --- maven-install-plugin:2.4:install (default-install) @ foo ---
...
[INFO] --- maven-site-plugin:3.12.0:site (default-site) @ foo ---
...
[INFO] >>> maven-surefire-report-plugin:2.22.2:report > [surefire]test @ foo >>>
[INFO]
[INFO] >>> maven-pmd-plugin:3.17.0:check (validate) > :pmd @ foo >>>
[INFO]
[INFO] --- maven-pmd-plugin:3.17.0:pmd (pmd) @ foo ---
[INFO] PMD version: 6.46.0
[INFO] 
[INFO] <<< maven-pmd-plugin:3.17.0:check (validate) < :pmd @ foo <<<
[INFO]
[INFO]
[INFO] --- maven-pmd-plugin:3.17.0:check (validate) @ foo ---

⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️
      LOOK AT THE FOLLOWING 3 LINES AND COMPARE: 
      THERE IS A VIOLATION MISSING!! 
⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️⚠️

[INFO] PMD version: 6.46.0
[INFO] PMD Failure: com.foo.ExampleClass:17 Rule:ImplicitSwitchFallThrough Priority:3 A switch statement does not contain a break.
[INFO] You have 1 PMD violation. For more details see: /[PROJECT-DIRECTORY]/target/pmd.xml
[INFO]
[INFO] >>> maven-pmd-plugin:3.17.0:cpd-check (validate) > :cpd @ foo >>>
[INFO]
[INFO] --- maven-pmd-plugin:3.17.0:cpd (cpd) @ foo ---
[INFO] PMD version: 6.46.0
[INFO] 
[INFO] <<< maven-pmd-plugin:3.17.0:cpd-check (validate) < :cpd @ foo <<<
[INFO]
[INFO]
[INFO] --- maven-pmd-plugin:3.17.0:cpd-check (validate) @ foo ---
[INFO] PMD version: 6.46.0
[INFO]
[INFO]
```

**What is happening:**
- The ``[PROJECT-DIRECTORY/target/pmd.xml]`` is being re-written after the tests execution, and one of the reported violations is missing. 
- An evidence of this is that the output console displays only 1 violation instead of 2, even though the missing one ('SwitchStmtsShouldHaveDefault', defined with priority 1) has a higher priority (given that 'ImplicitSwitchFallThrough' has priority 3 by default).
- Additionally, in the resulting HTML report (``[PROJECT-DIRECTORY]/target/site/index.html``) there is only 1 violation reported instead of the original 2 detected in the validation phase.

**Expected behavior:**
The PMD Maven plugin should report the same amount of violations detected when running ``mvn site`` as when running ``mvn install``, especially those that match the priority failure configuration.

**Running PMD through:** *Maven*

**Comment by UncleOwen** [comment]
With maven 3.8.7, maven-pmd-plugin 3.27.0, and pmd 7.16.0, this is what I get in target/pmd.xml when I run `mvn clean install site`:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<pmd xmlns="http://pmd.sourceforge.net/report/2.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://pmd.sourceforge.net/report/2.0.0 https://pmd.github.io/schema/report_2_0_0.xsd" version="7.16.0" timestamp="2025-08-28T14:05:42.095">
<file name="/data/code/pmd/repro-4086/src/main/java/com/example/SwitchExample.java">
<violation beginline="13" endline="21" begincolumn="9" endcolumn="10" rule="SwitchStmtsShouldHaveDefault" ruleset="Best Practices" package="com.example" class="SwitchExample" method="processType" externalInfoUrl="https://docs.pmd-code.org/pmd-doc-7.16.0/pmd_rules_java_bestpractices.html#nonexhaustiveswitch" priority="1">
Switch statements or expressions should be exhaustive, add a default case (or missing enum branches)
</violation>
<violation beginline="17" endline="17" begincolumn="13" endcolumn="26" rule="ImplicitSwitchFallThrough" ruleset="Error Prone" package="com.example" class="SwitchExample" method="processType" externalInfoUrl="https://docs.pmd-code.org/pmd-doc-7.16.0/pmd_rules_java_errorprone.html#implicitswitchfallthrough" priority="3">
This switch case may be reached by fallthrough from the previous case
</violation>
</file>
</pmd>
```

Does that mean this is fixed?
